### ISSUE TITLE
rol/CMakeLists.txt bug

### ISSUE THREAD

**Issue by jmgate** [trilinos/Trilinos#1488]
Just ran into a problem trying to configure Trilinos:  When you don't have `-D Trilinos_ENABLE_DEBUG:BOOL=ON` or `OFF` in your config script, @trilinos/rol's `CMakeLists.txt` file runs into a problem with

```
TRIBITS_ADD_OPTION_AND_DEFINE(${PACKAGE_NAME}_ENABLE_DEBUG
  HAVE_ROL_DEBUG
  "Enable a host of runtime debug checking."
  ${Trilinos_ENABLE_DEBUG}
  )
```

because `${Trilinos_ENABLE_DEBUG}` is empty, and `TRIBITS_ADD_OPTION_AND_DEFINE` therefore isn't getting the correct number of arguments.  The quick fix was to add `-D Trilinos_ENABLE_DEBUG:BOOL=OFF` to my config script, but it seems like this `CMakeLists.txt` file should be more robust than that.  You may be able to replace `${Trilinos_ENABLE_DEBUG}` with `${${PROJECT_NAME}_ENABLE_DEBUG}`, and things would just work, or you could check to make sure that last argument to `TRIBITS_ADD_OPTION_AND_DEFINE` is empty and replace it with `OFF` if it is.

Alternatively, this may be entirely user error&mdash;@trilinos/framework, does `-D Trilinos_ENABLE_DEBUG=?` *need* to be specified?  If so, my mistake.

**Comment by bartlettroscoe** [comment]
That is a defect in ROL.  I will file github issue and fix tomorrow.

-Ross
________________________________

**Comment by mhoemmen** [comment]
@jmgate Long answer is that Trilinos packages are supposed to use `TRIBITS_ADD_DEBUG_OPTION()` to add a debug option to their package, rather than making explicit use of `Trilinos_ENABLE_DEBUG`.

**Comment by bartlettroscoe** [comment]
Should be fixed as of commit 00fc3fab5980672 pushed last night.

@jmgate, please let me know if this fixes the issue for you (without having to set `-D Trilinos_ENABLE_DEBUG=<something>` )

**Comment by jmgate** [comment]
Yup, looks like that does the trick for me.  Thanks for the quick turn-around.
